src/templatefilters.py

```diff
@@ -47,6 +49,21 @@ def _create_search_link(m: Match):
         )
 
 
+def with_highlighted_term(s: str, content_regex: Optional[re.Pattern[Any]]):
+    if not s or content_regex is None:
+        return s
+
+    with_highlighted_terms = content_regex.sub(_create_highlighted_terms, s)
```

Comment:
If I'm understanding this correctly, the point of this is to build up a replacement string which has `{text}` replaced with the actual search text. I think we might be able to use backreferences to avoid the extra function here: https://stackoverflow.com/questions/8157267/handling-backreferences-to-capturing-groups-in-re-sub-replacement-pattern

Something like:
```
# app.py
if search_term:
    content_regex = re.compile(f"({search_term})", re.IGNORECASE)

# in templatefilters
with_highlighted_terms = re.sub(content_regex, r"<mark>(\1)</mark>", s)
```

I hope that all makes sense...that to say I don't think we need to format the replacement string, we can just use the `\1` reference!

Reply:
Ohhhh cool! Yeah that makes sense to me. I'll make the change, thanks!